package.json

```diff
@@ -44,13 +44,19 @@
   ],
   "scripts": {
     "toc": "doctoc README.md",
-    "lint": "(prettier . --check || true) && eslint .",
+    "lint": "prettier . --check && eslint .",
+    "lint:delta": "npm-run-all -p prettier:delta eslint:delta",
+    "prettier:delta": "prettier --check `./scripts/changed-files`",
+    "eslint:delta": "eslint `./scripts/changed-files`",
```

Comment:
I can understand the rationale behind incremental prettier, but incremental ESLint goes a step too far in my book. ESLint in our config has most of its formatting checks disabled via eslint-config-prettier, which means its mostly running code quality checks[^1] ("this should be a `const`", "var is not in scope," etc.)

Why go out of our way to avoid catching non-stylistic code quality issues?

[^1]: except for the import ordering plugin, but that doesn't feel very high risk for churn given its simplicity / consistency

Reply:
>I can understand the rationale behind incremental prettier, but incremental ESLint goes a step too far in my book.

Yes, you would do things differently. I got that the first time around.

Listen, we just went through that discussion. At this point in time, I will not have something that force changing large sway of code to appease automatic checks on formatting or linting. Nobody force you to use those delta scripts. You are quite welcome and encouraged to have new code conform to the lint checks. But, and I'll repeat again to be clear, at this point in time I will not have something that force changing large sway of code to appease automatic checks on formatting or linting.

Now, can we consider this particular discussion done, and move to other things?

Reply:
My apologies, I got a little unproductively worked up. I get a little passionate about testing practices 😅